### PR TITLE
WIP: Vol snap shot

### PR DESCRIPTION
**Pull request by vartikashah**
## Proposed changes

Add Volume snap shot [preview page](https://csc-guide-preview.rahtiapp.fi/origin/vol-snap-shot/cloud/rahti2/storage/).

https://csc-guide-preview.rahtiapp.fi/origin/vol-snap-shot/cloud/rahti2/storage/
https://csc-guide-preview.rahtiapp.fi/origin/vol-snap-shot/cloud/rahti2/storage/volume-snapshot/

## Checklist before requesting a review

- [x] I have followed the instructions in the [Contributing](https://github.com/CSCfi/csc-user-guide/blob/master/CONTRIBUTING.md) and [Styleguide](https://github.com/CSCfi/csc-user-guide/blob/master/STYLEGUIDE.md) documents.
- [x] My pull request passes the automated tests. (If not, visit [pull requests at Travis CI](https://app.travis-ci.com/github/CSCfi/csc-user-guide/pull_requests) and have a look at the job log.)
- [x] I have included a link to the appropriate [preview page](https://csc-guide-preview.rahtiapp.fi/origin/) (select your branch from the list).

### CODE REVIEW

**Review by jemaltahir — COMMENTED** [review]
Checkout the path it is under 
image
but supposed to be under like this:
image

File: docs/cloud/rahti2/storage/VolumeSnapshot.md
[the remark this answers is not in the document]
Comment (vartikashah) [reply docs/cloud/rahti2/storage/VolumeSnapshot.md:1]: done

[the remark this answers is not in the document]
Comment (jemaltahir) [reply docs/cloud/rahti2/storage/VolumeSnapshot.md:1]: Could you add a use case by deploying an Nginx application, creating content, taking a snapshot, and then restoring from that snapshot ?

File: docs/cloud/rahti2/storage/index.md [hunk 1]
```diff
@@ -23,3 +23,16 @@ In case you need to store large volumes of data, or you need that your data can
 At CSC, we offer Allas as our object storage solution. You can find additional information on the [Allas page](../../../data/Allas/index.md).
 
 You can find additional examples on how to backup to Allas [here](objectstorage.md).
+
+## Volume snapshots
+
+A snapshot represents the state of the storage volume in a cluster at a particular point in time. Volume snapshots can be used to provision a new volume. Volume snapshots help protect against data loss in OKD. Rahti2 support Container Storage Interface (CSI) volume snapshots by default and the default volume snapshot class name is standard-csi.
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/index.md:29]:
```suggestion
A snapshot represents the state of the storage volume within a cluster at a specific point in time. Volume snapshots can be used to provision new volume and help protect against data loss in OKD. Rahti2 support Container Storage Interface (CSI) volume snapshots by default and the default volume snapshot class name is standard-csi.
```

File: docs/cloud/rahti2/storage/index.md [hunk 2]
```diff
@@ -23,3 +23,16 @@ In case you need to store large volumes of data, or you need that your data can
...
+
+With CSI volume snapshots, an app developer can:
+
+- Use volume snapshots as building blocks for developing application or cluster-level storage backup solutions.
+- Rapidly rollback to a previous development version.
+- Use storage more efficiently by not having to make a full copy each time.
+
+You can find additional information on the [Volume snapshot](volume-snapshot.md) page.
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/index.md:37]:
```suggestion
- Use volume snapshots as building blocks for developing application-level or cluster-level storage backup solutions.
- Rapidly rollback to a previous version during development.
- Use storage more efficiently by avoiding the need to create a full copy each time.

You can find additional information on the [Volume snapshot](volume-snapshot.md) page.
```

File: docs/cloud/rahti2/storage/volume-snapshot.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
+## Volume snapshot provisioning
+
+There are two ways to provision snapshots in Rahti2: by web interface and by using CLI.
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/volume-snapshot.md:3]:
```suggestion
Rahti2 offers two methods for provisioning snapshots: through the web interface and by using the CLI.
```

File: docs/cloud/rahti2/storage/volume-snapshot.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+## Volume snapshot provisioning
+
+There are two ways to provision snapshots in Rahti2: by web interface and by using CLI. 
+
+# Prerequisites
+
+- An active project in Rahti2
+- No pods are using persistent volume claim (PVC) that you want to take a snapshot of.
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/volume-snapshot.md:8]:
```suggestion
- Ensure no pods are using persistent volume claim (PVC) that you want to take a snapshot of.
```

File: docs/cloud/rahti2/storage/volume-snapshot.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+1. Create a deployment
+2. Create a PVC
+3. Mount the PVC to deployment (in Rahti2 PVC is only created after that is mount to a deployment)
+4. Unmount the PVC from deployment
+5. Create a volume snapshot
+6. Attach the PVC to volume snapshot
+
+## Using web interface
+
+After making sure that PVC is not attached to any pod, create a volume snapshot by clicking on 'Create VolumeSnapshot' in 'VolumeSnapshot'.
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/volume-snapshot.md:21]:
```suggestion
After making sure that PVC is not attached to any pod. Navigate to the 'VolumeSnapshot' section and click on 'Create VolumeSnapshot' to create a snapshot of your PVC. 
```

File: docs/cloud/rahti2/storage/volume-snapshot.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+kind: VolumeSnapshot
+metadata:
+  name: <name_of_volumesnapshot>
+spec:
+  source:
+    persistentVolumeClaimName: <name_of_PVC>
+  volumeSnapshotClassName: standard-csi
+```
+
+To list all the volume snapshots, use the command:
```
Comment (jemaltahir) [inline docs/cloud/rahti2/storage/volume-snapshot.md:49]: Before list , it is good to show how to deploy the  snapshot.yaml? `oc apply -f  snapshot.yaml`

**Review by jemaltahir — CHANGES_REQUESTED** [review]
Have a look at the comments.

**Pull request closed without merging**